alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/AlfrescoStatusMetrics.java

```diff
@@ -0,0 +1,63 @@
+package eu.xenit.alfred.telemetry.binder;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import org.alfresco.repo.security.authentication.AuthenticationUtil;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.cmr.admin.RepoAdminService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+
+public class AlfrescoStatusMetrics implements MeterBinder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AlfrescoStatusMetrics.class);
+    private static final String STATUS_PREFIX = "alfresco.status";
+
+    private RepoAdminService repoAdminService;
+    private RetryingTransactionHelper retryingTransactionHelper;
+
+    public AlfrescoStatusMetrics(RepoAdminService repoAdminService, RetryingTransactionHelper retryingTransactionHelper) {
+        this.repoAdminService = repoAdminService;
+        this.retryingTransactionHelper = retryingTransactionHelper;
+    }
+
+    @Override
+    public void bindTo(@Nonnull MeterRegistry meterRegistry) {
+        LOGGER.info("Registering Alfresco Status metrics");
+        Gauge.builder(String.format("%s.%s", STATUS_PREFIX, "readonly"),
+                repoAdminService,
+                x -> getReadOnly(repoAdminService))
```

Comment:
The `ToDoubleFunction<T>` should consume the offered object. 

Hence: `x -> getReadOnly(x)` or with the method refence: `this::getReadOnly`

Reply:
The getReadOnly function returns a double.
ToDoubleFunction does not work on primitives, why would I use that?

Reply:
A function that has the `double` return type is a valid `ToDoubleFunction` .

Reply:
repoAdminService.getUsage().isReadOnly() returns a boolean

Reply:
`isReadOnly()` is irrelevant for this comment. `this` = an instance of `AlfrescoStatusMetrics`, which has the `getReadOnly(...)` method, that method returns a `double` and accepts a `RepoAdminService` and hence is a `ToDoubleFunction<RepoAdminService>`.

Reply:
Don't know what you mean, feel free to push changes if you want.

Reply:
FYI:
    @Override
    public void bindTo(@Nonnull MeterRegistry meterRegistry) {
        LOGGER.info("Registering Alfresco Status metrics");
        Gauge.builder(String.format("%s.%s", STATUS_PREFIX, "readonly"),
                repoAdminService, this::getReadOnly)
                .description("Metric about Alfresco being in read-only mode")
                .register(meterRegistry);
    }
    private double getReadOnly(RepoAdminService repoAdminService) {
        final boolean[] isReadOnly = {false};
        retryingTransactionHelper.doInTransaction(() -> {
            AuthenticationUtil.runAsSystem(() -> {
                isReadOnly[0] = repoAdminService.getUsage().isReadOnly();
                return null;
            });
            return null;
        },true);
        if(isReadOnly[0])
            return 1d;
        else
            return 0d;
    }

9:10
just try it
9:11
& ofcourse with the String.format("%s.%s", STATUS_PREFIX, "readonly")  changed :joy:
New
9:12
Otherwise we're running in circles :stuck_out_tongue: